### PR TITLE
llvmPackages_11: 11.0.0 -> 11.0.1

### PR DESCRIPTION
**Pull request by ggreif**
###### Motivation for this change

https://lists.llvm.org/pipermail/release-testers/2021-January/001436.html
http://lists.llvm.org/pipermail/release-testers/2020-December/001422.html
http://lists.llvm.org/pipermail/llvm-dev/2020-November/146911.html

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ggreif** [comment]
@GrahamcOfBorg build llvmPackages_11

**Comment by ggreif** [comment]
@lovek323 @7c6f434c @primeos I think this is good to go now.

File: pkgs/development/compilers/llvm/11/clang/default.nix [hunk 1]
```diff
@@ -8,12 +8,12 @@ let
...
     inherit version;
 
-    src = fetch "clang" "02ajkij85966vd150iy246mv16dsaph1kfi0y8wnncp8w6nar5hg";
+    src = fetch "clang" "0kab4zmkxffg98a3rx95756jlwhxflalin5w05g1anpwxv175xbk";
     inherit clang-tools-extra_src;
 
     unpackPhase = ''
       unpackFile $src
-      mv clang-${version}* clang
+      mv clang-* clang
```
Comment (primeos) [inline pkgs/development/compilers/llvm/11/clang/default.nix:16]:
```suggestion
      mv clang-${release_version}* clang
```

Probably doesn't matter but from your commit message:
> it was looking for `clang-11.0.1-rc2.src` but actually it is `clang-11.0.1rc2.src`

this should also work and would be consistent with your other change below.

File: pkgs/development/compilers/llvm/11/default.nix [hunk 2]
```diff
@@ -6,16 +6,18 @@
 }:
 
 let
-  release_version = "11.0.0";
-  version = release_version; # differentiating these (variables) is important for RCs
+  release_version = "11.0.1";
+  candidate = ""; # empty or "rcN"
+  dash-candidate = stdenv.lib.optionalString (builtins.stringLength candidate != 0) "-${candidate}";
```
Comment (primeos) [inline pkgs/development/compilers/llvm/11/default.nix:11]:
```suggestion
  dash-candidate = stdenv.lib.optionalString (candidate != "") "-${candidate}";
```

Shouldn't make a difference but is a bit more compact (nitpick).

**Review by primeos — COMMENTED** [review]
Thanks! I'll try to build ~~Chromium~~ Mesa and merge if it works.
(Edit: For some reasons I had to commit both suggestions separately but I'll squash and merge anyway.)

File: pkgs/development/compilers/llvm/11/clang/default.nix [hunk 3]
```diff
@@ -8,12 +8,12 @@ let
...
     inherit version;
 
-    src = fetch "clang" "02ajkij85966vd150iy246mv16dsaph1kfi0y8wnncp8w6nar5hg";
+    src = fetch "clang" "0kab4zmkxffg98a3rx95756jlwhxflalin5w05g1anpwxv175xbk";
     inherit clang-tools-extra_src;
 
     unpackPhase = ''
       unpackFile $src
-      mv clang-${version}* clang
+      mv clang-${release_version}* clang
```
Comment (primeos) [inline pkgs/development/compilers/llvm/11/clang/default.nix:16]:
```suggestion
      mv clang-* clang
```

My bad :o Not possible here.

**Comment by primeos** [comment]
@GrahamcOfBorg build mesa

**Review by primeos — APPROVED** [review]
LGTM, thanks!

(And sorry about the minor mess I've produced earlier, I was a bit tired...)

**Pull request merged**